src/platform/packages/shared/kbn-scout/src/playwright/fixtures/scope/worker/api_key.ts

```diff
@@ -77,63 +127,67 @@ export const requestAuthFixture = coreWorkerFixtures.extend<
           if (response.statusCode !== 200) {
             log.info(`Failed to invalidate API key: ${apiKey.name}`);
           } else {
-            log.info(`Successfully invalidated API key: ${apiKey.name}`);
+            log.info(`Invalidated API key: ${apiKey.name}`);
           }
         }
       };
 
-      const getApiKey = async (role: string): Promise<RoleApiCredentials> => {
-        const apiKeyName = `myTestApiKey-${generatedApiKeys.length}-worker-${
-          workerInfo.parallelIndex + 1
-        }`;
-        const adminCookieHeader = await samlAuth.session.getApiCredentialsForRole('admin');
-
+      const getApiKey = async (predefinedRole: string): Promise<RoleApiCredentials> => {
+        // fetch role descriptors from roles.yml file
         const roleDescriptors =
-          role === 'admin'
+          predefinedRole === 'admin'
             ? {}
             : (() => {
-                const roleDescriptor = defaultRoles.get(role);
+                const roleDescriptor = defaultRoles.availableRoles.get(predefinedRole);
                 if (!roleDescriptor) {
-                  throw new Error(`Cannot create API key for non-existent role "${role}"`);
+                  const availableRoles = Array.from(defaultRoles.availableRoles.keys()).join(', ');
```

Comment:
In the same vein of https://github.com/elastic/kibana/pull/243160, I've added a new explicative log message when using a non-existent predefined role:

```ts
Error: Role 'hello_api_test_role' not found in the supported list: /Users/cesaredecal/dev/kibana/src/platform/packages/shared/kbn-es/src/stateful_resources/roles.yml. Available predefined roles: viewer, editor, admin, system_indices_superuser.

Is 'hello_api_test_role' a custom test role? → Use getApiKeyForCustomRole() to log in with custom Kibana and Elasticsearch privileges

Is 'hello_api_test_role' a predefined role? (e.g., admin, viewer, editor) → Add the role descriptor to /Users/cesaredecal/dev/kibana/src/platform/packages/shared/kbn-es/src/stateful_resources/roles.yml to enable it for testing.
```